code/modules/projectiles/special.dm

```diff
@@ -674,6 +674,90 @@ ABSTRACT_TYPE(/datum/projectile/special)
 					boutput(dropme, __red("Your coffin was lost or destroyed! Oh no!!!"))
 		..()
 
+/datum/projectile/special/homing/magicmissile
+	name = "magic missile"
+	sname = "magic missile"
+	icon = 'icons/obj/wizard.dmi'
+	icon_state = "magicm"
+	shot_sound = null
+	power = 10
+	cost = 1
+	damage_type = D_KINETIC
+	dissipation_delay = 0
+	dissipation_rate = 0
+	ks_ratio = 1
+	brightness = 2
+	projectile_speed = 15
+	is_magical = 1 // It passes right through them, but just for consistency
+	auto_find_targets = 0
+	min_speed = 0.1
+	max_speed = 15
+	goes_through_walls = 0 // It'll stop homing when it hits something, then go bouncy
+	var/max_bounce_count = 3 // putting the I in ICEE BEEYEM
+	var/weaken_length = 5 SECONDS
+	var/impact_brute = 5
+	var/bouncy = 0 // A fallback for when a missile is shot without a target
+	var/slam_text = "The magic missile SLAMS into you!"
+	var/hit_sound = 'sound/effects/mag_magmisimpact_bounce.ogg'
+	var/cat_sound = 'sound/voice/animal/cat.ogg'
+	var/last_sound_time = 0
+
+	on_pre_hit(var/atom/hit, var/angle, var/obj/projectile/O)
+		if(..()) return TRUE
+		if(isliving(hit))
+			var/mob/living/M = hit
+			if (iswizard(M) || M.traitHolder?.hasTrait("training_chaplain"))
+				boutput(M, "The magic missile passes right through you!")
+				. = TRUE
+
+		/// Missiles home into their targets until they hit a wall. Then they forget their target and just bounce around
+		else if(src.homing_active && isturf(hit) && hit?.density)
+			src.homing_active = 0
+			src.bouncy = 1
```

Comment:
modifying state on the projectile datum while having multiple projectiles sharing the datum at once. This could lead to unexpected behavior

Reply:
After working on the flamer projectiles, I realize now how much a bad idea it is to mess with the datum while the projectile's in flight, and to work with the item instead. So, I did that!